### DISCUSSION TITLE
C++ Implementation using C++ Builder 10.3 (Formally Borland C++ Builder)

### DISCUSSION THREAD

**Discussion by smilima** [PlummersSoftwareLLC/Primes#232] [Ideas]
I have compiled the CPP code using the new version of C++ Builder.  I have not changed anything in the code - I just thought it would be interesting to see how the MS and C++ Builder compilers compare.  I didn't know if you want to add this into the main tree under the PrimeCPP folder?  I have compiled it for Win32 and Win64.

To edit this code, you can download a community version of C++ Builder at: https://www.embarcadero.com/products/cbuilder/starter/free-download

I am not affiliated with Embarcadero at all, I just have been using their tools since they were Borland back in the late 90's,

The code is under my Git account at:
https://github.com/smiliea/PrimeSieveCppBuilder

Thanks!

**Comment by smilima** [comment]
BTW, on my machine:

Visual Studio 64 Bit:
Passes: 5028, Time: 5.000000, Avg: 0.000994, Limit: 1000000, Count1: 78498, Count2: 78498, Valid: 1

C++ Builder 64 Bit:
Passes: 5784, Time: 5.000000, Avg: 0.000864, Limit: 1000000, Count1: 78498, Count2: 78498, Valid: 1

**Comment by rbergen** [comment]
As a general principle, we don't include binaries in repos, in particular when it concerns builds of the solutions in this repo. 
A Dockerfile that (also) builds and runs C++ solutions using C++ Builder could technically be added. However, inspection of the terms of use of the Embarcadero community license made us conclude that they are incompatible with our benchmarking setup. The primary concern is that we intentionally enable anybody to execute benchmark runs on any supported hardware of their choice.

@jimmckeeth, Embarcadero's Chief Developer Advocate, opened issue #199 to offer his help with addressing this same concern for a Delphi solution that was submitted. If that leads to a situation where we can indeed incorporate Delphi in our automated benchmark setup, then we may be able to investigate if that solution can also be applied to C++ Builder.
